### PR TITLE
Add keyed directive docs

### LINKED ISSUE #846: Add keyed directive to our built in directive documentation
The `keyed` directive ([link to generated docs](https://lit.dev/docs/api/directives/#keyed)) isn't documented on the [Built in directives](https://lit.dev/docs/templates/directives/) page.

### PR DESCRIPTION
**Pull request by justinfagnani**
Fixes #846

### CODE REVIEW

File: packages/lit-dev-content/site/docs/templates/directives.md [hunk 1]
```diff
@@ -1015,6 +1024,90 @@ The `cache` directive caches the generated DOM for a given expression and input
...
+@customElement('my-element')
+class MyElement extends LitElement {
+
+  @property()
+  userId?: string = '';
+
+  render() {
+    return html`
+      <div>
+        ${keyed(this.userId, html`<user-card .userId=${this.userId}></user-card>)}
```
Comment (AndrewJakubowicz) [inline packages/lit-dev-content/site/docs/templates/directives.md:1082]:
```suggestion
        ${keyed(this.userId, html`<user-card .userId=${this.userId}></user-card>`)}
```

File: packages/lit-dev-content/site/docs/templates/directives.md [hunk 2]
```diff
@@ -1015,6 +1024,90 @@ The `cache` directive caches the generated DOM for a given expression and input
...
+
+<table>
+<thead><tr><th></th><th></th></tr></thead>
+<tbody>
+<tr>
+<td class="no-wrap-cell vcenter-cell">Import</td>
+<td class="wide-cell">
+
+```js
+import {leyed} from 'lit/directives/keyed.js';
```
Comment (AndrewJakubowicz) [inline packages/lit-dev-content/site/docs/templates/directives.md:1040]:
```suggestion
import {keyed} from 'lit/directives/keyed.js';
```

File: packages/lit-dev-content/site/docs/templates/directives.md [hunk 3]
```diff
@@ -1015,6 +1024,90 @@ The `cache` directive caches the generated DOM for a given expression and input
...
+
+  constructor() {
+    super();
+    this.userId = '';
+  }
+
+  render() {
+    return html`
+      <div>
+        ${keyed(this.userId, html`<user-card .userId=${this.userId}></user-card>)}
```
Comment (AndrewJakubowicz) [inline packages/lit-dev-content/site/docs/templates/directives.md:1102]:
```suggestion
        ${keyed(this.userId, html`<user-card .userId=${this.userId}></user-card>`)}
```

File: packages/lit-dev-content/site/docs/templates/directives.md [hunk 4]
```diff
@@ -1015,6 +1024,90 @@ The `cache` directive caches the generated DOM for a given expression and input
...
+`keyed` is also useful in some animation scenarios if you need to force a new element for "enter" or "exit" animations.
+
+{% switchable-sample %}
+
+```ts
+@customElement('my-element')
+class MyElement extends LitElement {
+
+  @property()
+  userId?: string = '';
```
Comment (AndrewJakubowicz) [inline packages/lit-dev-content/site/docs/templates/directives.md:1077]: No action required - Can the `?` be removed?

File: packages/lit-dev-content/site/docs/templates/directives.md [hunk 5]
```diff
@@ -122,6 +122,15 @@ Lit includes a number of built-in directives to help with a variety of rendering
...
   <td>Caches rendered DOM when changing templates rather than discarding the DOM.</td>
+  </tr>
+
+    <tr>
+  <td>
+
+  [`keyed`](#keyed)
+
+  </td>
+  <td>Associates a renderable value with a unique key, forcing the DOM to re-render if the key changes.</td>
```
Comment (AndrewJakubowicz) [inline packages/lit-dev-content/site/docs/templates/directives.md:133]: Something here has caused this artifact:
Screen Shot 2022-05-20 at 9 22 48 AM

The tags look completely balanced so maybe it's spacing?

**Review by AndrewJakubowicz — COMMENTED** [review]
The contents looks awesome! Some typo nits and a stray `<tr>` tag has jumped out of the table.

File: packages/lit-dev-content/site/docs/templates/directives.md [hunk 6]
```diff
@@ -122,6 +122,15 @@ Lit includes a number of built-in directives to help with a variety of rendering
 
   </td>
   <td>Caches rendered DOM when changing templates rather than discarding the DOM.</td>
+  </tr>
+
+    <tr>
```
Comment (arthurevans) [inline packages/lit-dev-content/site/docs/templates/directives.md:127]:
```suggestion
<tr>
```

File: packages/lit-dev-content/site/docs/templates/directives.md (hunk 5 above)
Comment (justinfagnani) [reply packages/lit-dev-content/site/docs/templates/directives.md]: That was some spacing that @arthurevans pointed out. Fixed.

**Review by AndrewJakubowicz — APPROVED** [review]
Nice! Also checked manually in the preview url and checked that `keyed` is searchable.

File: packages/lit-dev-content/site/docs/templates/directives.md [hunk 7]
```diff
@@ -1015,6 +1024,90 @@ The `cache` directive caches the generated DOM for a given expression and input
 
 Explore `cache` more in the [playground](/playground/#sample=examples/directive-cache).
 
+
+### keyed
+
+Associates a renderable value with a unique key. When the key changes, the previous DOM is removed and disposed before rendering the next value, even if the value - such as a template - is the same.
```
Comment (arthurevans) [inline packages/lit-dev-content/site/docs/templates/directives.md:1030]: For consistency—although I know you don't love it—use the em dash with no spaces. (If there was an easy way to get markdown-it or eleventy to render these as en dashes with narrow spaces, I'd be down for that, I just don't want to try and insert narrow spaces all over the place by hand.)

```suggestion
Associates a renderable value with a unique key. When the key changes, the previous DOM is removed and disposed before rendering the next value, even if the value—such as a template—is the same.
```

File: packages/lit-dev-content/site/docs/templates/directives.md [hunk 8]
```diff
@@ -1015,6 +1024,90 @@ The `cache` directive caches the generated DOM for a given expression and input
...
+<td class="wide-cell">
+
+Any expression
+
+</td>
+</tr>
+</tbody>
+</table>
+
+`keyed` is useful when rendering stateful elements and you need to ensure that all state of the element is cleared when some critical data changes. It essentially opts-out of Lit's default DOM reuse strategy.
```
Comment (arthurevans) [inline packages/lit-dev-content/site/docs/templates/directives.md:1066]:
```suggestion
`keyed` is useful when you're rendering stateful elements and you need to ensure that all state of the element is cleared when some critical data changes. It essentially opts-out of Lit's default DOM reuse strategy.
```

**Review by arthurevans — APPROVED** [review]
Two suggestions, otherwise GTG.

**Pull request merged**